entc/gen/template/dialect/sql/create.tmpl

```diff
@@ -180,7 +180,11 @@ func ({{ $receiver }} *{{ $builder }}) Save(ctx context.Context) ([]*{{ $.Name }
 				{{- if $.HasOneFieldID }}
 					mutation.{{ $.ID.BuilderField }} = &nodes[i].{{ $.ID.StructField }}
 					{{- if or $.ID.IsString $.ID.IsUUID $.ID.IsBytes $.ID.IsOther }}
-						{{- /* Do nothing, because these 2 types must be supplied by the user. */ -}}
+						{{- /* Do nothing, because these 4 types must be supplied by the user. */ -}}
+					{{- else if or $.ID.Type.ValueScanner (not $.ID.Type.Numeric) }}
```

Comment:
Why do you check `not $.ID.Type.Numeric`? Isn't the first check enough (i.e. `$.ID.Type.ValueScanner`)?

Reply:
Thanks!
Yes, `$.ID.Type.ValueScanner` only is enough.